src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/ImportAzureRmDataLakeStoreItem.cs

```diff
@@ -82,49 +85,113 @@ public int ConcurrentFileCount
             HelpMessage = "Indicates that, if the file or folder exists, it should be overwritten")]
         public SwitchParameter Force { get; set; }
 
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false,
+            HelpMessage = "Optionally indicates the diagnostic log level to use to record events during the file or folder import. Default is none, and specifying -Debug overwrites this value and sets it to Debug.")]
+        public LogLevel DiagnosticLogLevel
+        {
+            get { return logLevel; }
+            set { logLevel = value; }
+        }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false,
+            HelpMessage = "Optionally specifies the path for the diagnostic log to record events to during the file or folder import. If logging is enabled, the default is in %LOCALAPPDATA%\\ADLDataTransfer.")]
+        [ValidateNotNullOrEmpty]
+        public string DiagnosticLogPath { get; set; }
+
         public override void ExecuteCmdlet()
         {
             var powerShellSourcePath = SessionState.Path.GetUnresolvedProviderPathFromPSPath(Path);
+            string diagnosticPath = string.Empty;
+            if (!string.IsNullOrEmpty(DiagnosticLogPath))
+            {
+                diagnosticPath = SessionState.Path.GetUnresolvedProviderPathFromPSPath(DiagnosticLogPath);
+            }
+
+            // Setting -Debug overwrites LogLevel to debug.
+            if (MyInvocation.BoundParameters.ContainsKey("Debug") && (SwitchParameter)MyInvocation.BoundParameters["Debug"])
```

Comment:
Fair enough. I have removed anything involving the -Debug flag for now, and I agree that it would be nice to have some way to enable some kind of logging (to a file or otherwise) with the -debug flag, or be able to use some other kind of interceptor. We can discuss this more during PowerShell office hours, since I would like to do something that can benefit everyone and not just these two cmdlets (ideally!)